File: RudyAIR/src/main/java/rudyAir/model/vol/Vol.java
package rudyAir.model.vol;

import java.time.LocalDate;
import java.util.List;
import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Version;

import com.fasterxml.jackson.annotation.JsonView;

import rudyAir.model.Views;
import rudyAir.model.compte.Reservation;

@Entity
@Table(name = "vol")
@SequenceGenerator(name = "seqVol", sequenceName = "seq_vol", initialValue = 100, allocationSize = 1)
public class Vol {

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqVol")
	@Column(name = "vol_id")
	@JsonView({Views.Common.class, Views.ReservationPassagerWithVol.class})
	private Long id;
	@Column(name = "vol_date_depart")
	@JsonView({Views.ReservationPassagerWithVol.class})
	private LocalDate dateDepart;
	@Column(name = "vol_date_arrivee")
	@JsonView({Views.ReservationPassagerWithVol.class})
	private LocalDate dateArrivee;
	@Enumerated(EnumType.STRING)
	@Column(name = "vol_statut")
	@JsonView(Views.Common.class)
	private StatutVol statutVol;
	@Column(name = "vol_numero")
	@JsonView(Views.Common.class)
	private String numeroVol;
	@OneToOne
	@JoinColumn(name = "vol_volGen_id", foreignKey = @ForeignKey(name = "vol_volGen_id_fk"))
	private VolGenerique volGenerique;
	@OneToOne
	@JoinColumn(name = "vol_avion_id", foreignKey = @ForeignKey(name = "vol_avion_id_fk"))
	private Avion avion;
	@OneToMany(mappedBy = "vol")
	private List<Reservation> reservations;
	@Version
	private int version;

	public Vol() {
	}

	public Vol(Long id, LocalDate dateDepart, LocalDate dateArrivee, StatutVol statutVol, String numeroVol,
			VolGenerique volGenerique, Avion avion, List<Reservation> reservations) {
		super();
		this.id = id;
		this.dateDepart = dateDepart;
		this.dateArrivee = dateArrivee;
		this.statutVol = statutVol;
		this.numeroVol = numeroVol;
		this.volGenerique = volGenerique;
		this.avion = avion;
		this.reservations = reservations;
	}

	public Vol(LocalDate dateDepart, LocalDate dateArrivee, StatutVol statutVol, String numeroVol,
			VolGenerique volGenerique, Avion avion) {
		super();
		this.dateDepart = dateDepart;
		this.dateArrivee = dateArrivee;
		this.statutVol = statutVol;
		this.numeroVol = numeroVol;
		this.volGenerique = volGenerique;
		this.avion = avion;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public LocalDate getDateArrivee() {
		return dateArrivee;
	}

	public void setDateArrivee(LocalDate dateArrivee) {
		this.dateArrivee = dateArrivee;
	}

	public LocalDate getDateDepart() {
		return dateDepart;
	}

	public void setDateDepart(LocalDate dateDepart) {
		this.dateDepart = dateDepart;
	}

	public StatutVol getStatutVol() {
		return statutVol;
	}

	public void setStatutVol(StatutVol statutVol) {
		this.statutVol = statutVol;
	}

	public String getNumeroVol() {
		return numeroVol;
	}

	public void setNumeroVol(String numeroVol) {
		this.numeroVol = numeroVol;
	}

	public VolGenerique getVolGenerique() {
		return volGenerique;
	}

	public void setVolGenerique(VolGenerique volGenerique) {
		this.volGenerique = volGenerique;
	}

	public Avion getAvion() {
		return avion;
	}

	public void setAvion(Avion avion) {
		this.avion = avion;
	}

	public List<Reservation> getReservations() {
		return reservations;
	}

	public void setReservations(List<Reservation> reservations) {
		this.reservations = reservations;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Vol other = (Vol) obj;
		return Objects.equals(id, other.id);
	}

}

File: rudyAirAngular/src/app/model/vol/vol.ts
import { Avion } from './avion';
import { StatutVol } from './statut-vol';

export class Vol {
  private _id: number | undefined;
  private _numeroVol: string | undefined;
  private _dateDepart: Date | undefined;
  private _dateArrivee: Date | undefined;
  private _statutVol: StatutVol | undefined;
  private _volGenerique: VolGenerique | undefined;
  private _avion: Avion | undefined;
  private _reservation: Reservation | undefined;

  constructor(
    id?: number,
    numeroVol?: string,
    dateDepart?: Date,
    dateArrivee?: Date,
    statutVol?: StatutVol,
    volGenerique?: VolGenerique,
    avion?: Avion,
    reservation?: Reservation
  ) {
    this._id = id;
    this._numeroVol = numeroVol;
    this._dateDepart = dateDepart;
    this._dateArrivee = dateArrivee;
    this._statutVol = statutVol;
    this._volGenerique = volGenerique;
    this._avion = avion;
    this._reservation = reservation;
  }

  /**
   * Getter id
   * @return {number }
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter numeroVol
   * @return {string }
   */
  public get numeroVol(): string | undefined {
    return this._numeroVol;
  }

  /**
   * Getter dateDepart
   * @return {Date }
   */
  public get dateDepart(): Date | undefined {
    return this._dateDepart;
  }

  /**
   * Getter dateArrivee
   * @return {Date }
   */
  public get dateArrivee(): Date | undefined {
    return this._dateArrivee;
  }

  /**
   * Getter statutVol
   * @return {StatutVol }
   */
  public get statutVol(): StatutVol | undefined {
    return this._statutVol;
  }

  /**
   * Getter volGenerique
   * @return {VolGenerique }
   */
  public get volGenerique(): VolGenerique | undefined {
    return this._volGenerique;
  }

  /**
   * Getter avion
   * @return {Avion }
   */
  public get avion(): Avion | undefined {
    return this._avion;
  }

  /**
   * Getter reservation
   * @return {Reservation }
   */
  public get reservation(): Reservation | undefined {
    return this._reservation;
  }

  /**
   * Setter id
   * @param {number } value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter numeroVol
   * @param {string } value
   */
  public set numeroVol(value: string | undefined) {
    this._numeroVol = value;
  }

  /**
   * Setter dateDepart
   * @param {Date } value
   */
  public set dateDepart(value: Date | undefined) {
    this._dateDepart = value;
  }

  /**
   * Setter dateArrivee
   * @param {Date } value
   */
  public set dateArrivee(value: Date | undefined) {
    this._dateArrivee = value;
  }

  /**
   * Setter statutVol
   * @param {StatutVol } value
   */
  public set statutVol(value: StatutVol | undefined) {
    this._statutVol = value;
  }

  /**
   * Setter volGenerique
   * @param {VolGenerique } value
   */
  public set volGenerique(value: VolGenerique | undefined) {
    this._volGenerique = value;
  }

  /**
   * Setter avion
   * @param {Avion } value
   */
  public set avion(value: Avion | undefined) {
    this._avion = value;
  }

  /**
   * Setter reservation
   * @param {Reservation } value
   */
  public set reservation(value: Reservation | undefined) {
    this._reservation = value;
  }
}
